core/src/commonMain/kotlin/com/ustadmobile/core/controller/PersonEditPresenter.kt

```diff
@@ -55,6 +71,22 @@ class PersonEditPresenter(context: Any,
                 MessageIdOption(MessageID.male, context, Person.GENDER_MALE),
                 MessageIdOption(MessageID.other, context, Person.GENDER_OTHER))
         view.clazzList = clazzMemberJoinEditHelper.liveList
+        val mWorkSpace = arguments[UstadView.ARG_WORKSPACE]
+        if(mWorkSpace != null){
+            workSpace = Json.parse(WorkSpace.serializer(), mWorkSpace)
+        }
+
+        serverUrl = if (arguments.containsKey(UstadView.ARG_SERVER_URL)) {
```

Comment:
I think the serverUrl should be required if it is in registration mode. The serverUrl should be nullable as it would be null if it is not in registration mode.